daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/ExpressionEvaluatingParsers.scala

```diff
@@ -158,25 +158,24 @@ class SetVariableParser(expr: CompiledExpression[AnyRef], decl: VariableRuntimeD
   }
 }
 
-class NewVariableInstanceStartParser(
-  override val context: RuntimeData)
+class NewVariableInstanceStartParser(override val context: VariableRuntimeData)
   extends PrimParser {
-  context.notYetImplemented("newVariableInstance")
   override lazy val runtimeDependencies = Vector()
 
-  def parse(pstate: PState) = {
-    context.notYetImplemented("newVariableInstance")
+  def parse(start: PState): Unit = {
+    log(LogLevel.Debug, "This is %s", toString) // important. Don't toString unless we have to log.
+    if (start.processorStatus.isInstanceOf[Failure]) return
+    // val vmap = start.variableMap
+    start.newVariableInstance(context)
```

Comment:
In what case does the previous processor fail but we continue to call this one? If we do need thi heck, can you remove the return, so it's something like
```
if (start.processorStatus == Success) {
  start.newVariableInstance(contxt)
}
```

Can you remove the comment.